backend/src/server/routes/v3/dashboard-router.ts

```diff
@@ -173,110 +174,115 @@ export const registerDashboardRouter = async (server: FastifyZodProvider) => {
-                  type: EventType.GET_SECRETS,
-                  metadata: {
-                    environment,
-                    secretPath,
-                    numberOfSecrets: secretCountFromEnv
-                  }
-                }
-              });
-
-              if (getUserAgentType(req.headers["user-agent"]) !== UserAgentType.K8_OPERATOR) {
-                await server.services.telemetry.sendPostHogEvents({
-                  event: PostHogEventTypes.SecretPulled,
-                  distinctId: getTelemetryDistinctId(req),
-                  properties: {
-                    numberOfSecrets: secretCountFromEnv,
-                    workspaceId: projectId,
-                    environment,
-                    secretPath,
-                    channel: getUserAgentType(req.headers["user-agent"]),
-                    ...req.auditLogInfo
+          if (remainingLimit > 0 && totalSecretCount > adjustedOffset) {
+            secrets = await server.services.secret.getSecretsRawMultiEnv({
+              actorId: req.permission.id,
+              actor: req.permission.type,
+              actorOrgId: req.permission.orgId,
+              environments,
+              actorAuthMethod: req.permission.authMethod,
+              projectId,
+              path: secretPath,
+              orderBy,
+              orderDirection,
+              search,
+              limit: remainingLimit,
+              offset: adjustedOffset
+            });
+
+            for await (const environment of environments) {
+              const secretCountFromEnv = secrets.filter((secret) => secret.environment === environment).length;
+
+              if (secretCountFromEnv) {
+                await server.services.auditLog.createAuditLog({
+                  projectId,
+                  ...req.auditLogInfo,
+                  event: {
+                    type: EventType.GET_SECRETS,
+                    metadata: {
+                      environment,
+                      secretPath,
+                      numberOfSecrets: secretCountFromEnv
+                    }
                   }
                 });
+
+                if (getUserAgentType(req.headers["user-agent"]) !== UserAgentType.K8_OPERATOR) {
+                  await server.services.telemetry.sendPostHogEvents({
+                    event: PostHogEventTypes.SecretPulled,
+                    distinctId: getTelemetryDistinctId(req),
+                    properties: {
+                      numberOfSecrets: secretCountFromEnv,
+                      workspaceId: projectId,
+                      environment,
+                      secretPath,
+                      channel: getUserAgentType(req.headers["user-agent"]),
+                      ...req.auditLogInfo
+                    }
+                  });
+                }
               }
             }
           }
         }
+      } catch (error) {
+        if (!(error instanceof ForbiddenError)) {
+          throw error;
+        }
```

Comment:
so if it's forbidden, we don't throw any error? why?

Reply:
Have explained in today's chat in group also have did the same in the other PR for this one

https://github.com/Infisical/infisical/pull/2488#discussion_r1776902807

Reply:
ah okay gotcha!